src/internal/components/pluggableVisualizations/xirr/tests/PluggableXirr.spec.tsx

```diff
@@ -0,0 +1,223 @@
+                .mockImplementation(() => fakeElement);
+            const reactRenderSpy = jest.spyOn(ReactDom, "render").mockImplementation(jest.fn());
+
+            const headline = createComponent({
+                featureFlags: {
+                    disableKpiDashboardHeadlineUnderline: true,
+                },
+            });
+
+            const options: IVisProps = getTestOptions();
+
+            headline.update(options, null, testMocks.emptyMdObject);
+
+            expect(reactCreateElementSpy.mock.calls[0][0]).toBe(Xirr);
+            expect(reactCreateElementSpy.mock.calls[0][1]).toEqual(
+                createExpectedProps(options, {
+                    config: {
+                        disableDrillUnderline: true,
+                    },
+                }),
+            );
+
+            reactCreateElementSpy.mockReset();
+            reactRenderSpy.mockReset();
+        });
+    });
+
+    describe("getExtendedReferencePoint", () => {
+        it("should return proper extended reference point", async () => {
+            const extendedReferencePoint = await createComponent().getExtendedReferencePoint(
+                referencePointMocks.measuresAndDateReferencePoint,
+            );
+
+            const expectedBuckets: IBucket[] = [
+                {
+                    localIdentifier: "measures",
+                    items: referencePointMocks.measuresAndDateReferencePoint.buckets[0].items.slice(0, 1),
+                },
+                {
+                    localIdentifier: "attribute",
+                    items: referencePointMocks.measuresAndDateReferencePoint.buckets[1].items.slice(0, 1),
+                },
+            ];
+
+            const expectedFilters: IFilters = {
+                localIdentifier: "filters",
+                items: [],
+            };
+
+            expect(extendedReferencePoint).toEqual({
+                buckets: expectedBuckets,
+                filters: expectedFilters,
+                properties: {},
+                uiConfig: uiConfigMocks.fullySpecifiedXirrUiConfig,
+            });
+        });
+
+        it("should correctly process empty reference point", async () => {
+            const headline = createComponent();
+            const extendedReferencePoint = await headline.getExtendedReferencePoint(
+                referencePointMocks.emptyReferencePoint,
+            );
+
+            const expectedBuckets: IBucket[] = [
+                {
+                    localIdentifier: "measures",
+                    items: [],
+                },
+                {
+                    localIdentifier: "attribute",
+                    items: [],
+                },
+            ];
+
+            const expectedFilters: IFilters = {
+                localIdentifier: "filters",
+                items: [],
+            };
+
+            const expectedUiConfig = { ...uiConfigMocks.fullySpecifiedXirrUiConfig };
+            expectedUiConfig.buckets.measures.canAddItems = true;
```

Comment:
I am pretty sure you are modifying the exported object here.

Reply:
Yes, that is intentional. I want to use the `...uiConfigMocks.fullySpecifiedXirrUiConfig` but want to change two relatively deeply nested properties. It seemed better than lots of `...`

Reply:
Oh, I see your point now, yes that is probably wrong, cloned just to be sure. Good catch!